app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestService.kt

```diff
@@ -67,52 +70,66 @@ class MeetingRequestService : FirebaseMessagingService() {
   override fun onMessageReceived(remoteMessage: RemoteMessage) {
     super.onMessageReceived(remoteMessage)
     val senderUid = remoteMessage.data["senderUID"] ?: "null"
-    val message = remoteMessage.data["message"] ?: "null"
     val title = remoteMessage.data["title"] ?: "null"
     val senderName = remoteMessage.data["senderName"] ?: "null"
 
     when (title) {
       "MEETING REQUEST" -> {
+        val message1 = remoteMessage.data["message1"] ?: "null"
+        val message2 = remoteMessage.data["message2"] ?: "null"
+        val locationString = remoteMessage.data["location"] ?: "null"
+        val latitudeString = locationString.split(", ")[0]
+        val longitudeString = locationString.split(", ")[1]
+        val location = (latitudeString.toDouble() to longitudeString.toDouble())
         MeetingRequestManager.meetingRequestViewModel?.addToMeetingRequestInbox(
-            senderUid, message) {
+            senderUid, message1, message2, location) {
               MeetingRequestManager.meetingRequestViewModel?.updateInboxOfMessages {}
             }
-        showNotification(MSG_REQUEST, "from : $senderName")
+        if (!isAppInForeground()) {
+          showNotification(MSG_REQUEST, "from : $senderName")
+        }
```

Comment:
Nice way to handle if the user is on the app or not.

Reply:
thank you !